src/core/mount.c

```diff
@@ -1354,16 +1355,37 @@ static int mount_stop(Unit *u) {
                 mount_enter_signal(m, MOUNT_UNMOUNTING_SIGKILL, MOUNT_SUCCESS);
                 return 0;
 
-        case MOUNT_DEAD:
-        case MOUNT_FAILED:
-                /* The mount has just been unmounted by somebody else. */
-                return 0;
-
         default:
                 assert_not_reached();
         }
 }
 
+int mount_stop_by_path(Manager *manager, const char *path) {
+        _cleanup_free_ char *name = NULL;
+        Unit *u;
+        int r;
+
+        assert(manager);
+        assert(path);
+
+        /* This returns 1 when the path will be unmounted, 0 if the path is already unmounted,
+         * negative errno otherwise. */
+
+        r = unit_name_from_path(path, ".mount", &name);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to generate unit name from path \"%s\", ignoring: %m", path);
+
+        u = manager_get_unit(manager, name);
+        if (!u)
+                return 0; /* Not necessary to unmount the path. */
+
+        if (unit_active_state(u) == UNIT_DEACTIVATING)
+                return 1; /* We are already unmounting the path. Note, this check is necessary, as
+                           * unit_stop() will return 0 when the unit is already in unmounting state. */
+
+        return unit_stop(u);
```

Comment:
Hmm, I'm not sure about calling `unit_stop` outside of job logic? Existing code always use `manager_add_job`?